src/metabase/upload.clj

```diff
@@ -277,16 +277,11 @@
    The value of `extant-columns` and `generated-columns` is an ordered map of normalized-column-name -> type for the
    given CSV file. Supported types include `::int`, `::datetime`, etc. A column that is completely blank is assumed to
    be of type `::text`."
-  [header rows]
-  (let [normalized-header (->> header
-                               (map normalize-column-name))
-        unique-header     (->> normalized-header
-                               mbql.u/uniquify-names
-                               (map keyword))
-        column-count      (count normalized-header)
-        settings          (upload-parsing/get-settings)
-        col-name+type-pairs (->> rows
-                                 (column-types-from-rows settings column-count)
+  [settings header rows]
+  (let [normalized-header   (map normalize-column-name header)
+        unique-header       (map keyword (mbql.u/uniquify-names normalized-header))
+        column-count        (count normalized-header)
+        col-name+type-pairs (->> (column-types-from-rows settings column-count rows)
```

Comment:
Sorry for the line noise here - this code so so nearby that it got caught in the cross-fire

IMHO this code  is easier to read without it so much threading, or at least I applied my personal conventions:

1. No degenerate first form - we read  s-expressions all day.
2. Never thread through just 1 form if you could rather have a 1-liner without dense nesting.

Reply:
There was a reason for this style: If you squint you can see `normalized-header` and `unique-header` as one big transformation, with `normalized-header` being an intermediate binding in the transformation. It made sense at the time anyway :)

Reply:
I'm okay with this change though, it's obviously not everyone's cup of tea